test/iterator.js

```diff
@@ -643,6 +643,41 @@ metatests.testSync('Iterator.chainApply on string', test => {
   test.strictSame(actual, 'h e l l o');
 });
 
+metatests.testSync('Iterator.chunk', test => {
+  const actual = iter([1, 2, 3, 1, 4, 5, 1, 6, 7])
+    .chunk(v => v === 1)
+    .toArray();
+  test.strictSame(actual, [[1, 2, 3], [1, 4, 5], [1, 6, 7]]);
+});
+
+metatests.testSync('Iterator.chunk with number', test => {
+  const actual = iter([1, 2, 3, 1, 4, 5, 1, 6, 7])
+    .chunk(3)
+    .toArray();
+  test.strictSame(actual, [[1, 2, 3], [1, 4, 5], [1, 6, 7]]);
+});
+
+metatests.testSync('Iterator.chunk with empty iterator', test => {
+  const actual = iter([])
+    .chunk(3)
+    .toArray();
+  test.strictSame(actual, []);
+});
+
+metatests.testSync('Iterator.chunk with single chunk', test => {
+  const actual = iter([1, 2, 3, 1, 4, 5, 1, 6, 7])
+    .chunk(() => false)
```

Comment:
Can you also add an example with `.chunk(() => true)`?

Reply:
Thanks, this caught a bug where the last chunk could have been ignored.